src/transformStateWithClones.js

```diff
@@ -7,7 +7,25 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const resultArray = [];
+  let copyState = { ...state };
+
+  for (const action of actions) {
+    if (action.type === 'addProperties') {
+      Object.assign(copyState, action.extraData);
+      resultArray.push(Object.assign({}, copyState));
+    } else if (action.type === 'removeProperties') {
+      for (const key of action.keysToRemove) {
+        delete copyState[key];
+      }
+      resultArray.push(Object.assign({}, copyState));
+    } else if (action.type === 'clear') {
+      copyState = {};
+      resultArray.push({});
```

Comment:
[DONT REPEAT YOURSELF]: Again, move the push to `resultArray` outside of the `if...else if` structure to avoid repetition and perform this action only once per loop iteration.